src/routes/user.ts

```diff
@@ -0,0 +1,88 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { Type, Static } from '@sinclair/typebox';
+import { knexController } from '../database/database';
+
+const userAddedResponse = Type.Object({
+  status: Type.String(),
+});
+
+export type UserAddedResponse = Static<typeof userAddedResponse>;
+
+const schema = {
+  description: 'Add new user or return error if add is not successful',
+  summary: 'Add user',
+  tags: ['User'],
+  body: {
+    type: 'object',
+    required: ['email', 'forename', 'surname', 'password'],
+    properties: {
+      email: {
+        type: 'string',
+      },
+      forename: {
+        type: 'string',
+      },
+      surname: {
+        type: 'string',
+      },
+      password: {
+        type: 'string',
+      },
+    },
+  },
```

Comment:
Tän vois varmaan tyypittää typeboxilla? Voi olla kiva, että muuallakin ohjelmassa voitaisiin käyttää esim `CreateUser` -tyyppiä, varsinkin jos jossain kohtaa aletaan exportoimaan tyyppejä bäkkäriltä frontille.

Reply:
https://github.com/sinclairzx81/typebox#usage

Hyvä esimerkki, että typebox oksentaa juuri tuollaisia Ajv yhteensopivia schemoja. Fastify tosiaan käyttää Ajv v8 requestien validointiin

https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#core-concepts

Reply:
> ...Hyvä esimerkki, että typebox oksentaa juuri tuollaisia Ajv yhteensopivia schemoja. Fastify tosiaan käyttää Ajv v8 requestien validointiin
> 
> https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#core-concepts

Joo tiiän tän